app/models/task/Task.scala

```diff
@@ -109,59 +109,90 @@ class TaskDAO @Inject()(sqlClient: SQLClient, projectDAO: ProjectDAO)(implicit e
     s"""
-        select ${columnsWithPrefix("webknossos.tasks_.")}
-           from
-             webknossos.tasks_
-             join
-               (select domain, value
-                from webknossos.user_experiences
-                where _user = '${userId.id}')
-               as user_experiences on webknossos.tasks_.neededExperience_domain = user_experiences.domain and webknossos.tasks_.neededExperience_value <= user_experiences.value
-             join webknossos.projects_ on webknossos.tasks_._project = webknossos.projects_._id
-             left join (select _task from webknossos.annotations_ where _user = '${userId.id}' and typ = '${AnnotationType.Task}' and not ($isTeamManagerOrAdmin and state = '${AnnotationState.Cancelled}')) as userAnnotations ON webknossos.tasks_._id = userAnnotations._task
-           where webknossos.tasks_.openInstances > 0
-                 and webknossos.projects_._team in ${writeStructTupleWithQuotes(teamIds.map(t => sanitize(t.id)))}
-                 and userAnnotations._task is null
-                 and not webknossos.projects_.paused
-           order by webknossos.projects_.priority desc
-           limit 1
-      """
+          select ${columnsWithPrefix("webknossos.tasks_.")}
+             from
+               webknossos.tasks_
+               join
+                 (select domain, value
+                  from webknossos.user_experiences
+                  where _user = '${userId.id}')
+                 as user_experiences on webknossos.tasks_.neededExperience_domain = user_experiences.domain and webknossos.tasks_.neededExperience_value <= user_experiences.value
+               join webknossos.projects_ on webknossos.tasks_._project = webknossos.projects_._id
+               left join (select _task from webknossos.annotations_ where _user = '${userId.id}' and typ = '${AnnotationType.Task}' and not ($isTeamManagerOrAdmin and state = '${AnnotationState.Cancelled}')) as userAnnotations ON webknossos.tasks_._id = userAnnotations._task
+             where webknossos.tasks_.openInstances > 0
+                   and webknossos.projects_._team in ${writeStructTupleWithQuotes(teamIds.map(t => sanitize(t.id)))}
+                   and userAnnotations._task is null
+                   and not webknossos.projects_.paused
+             order by webknossos.projects_.priority desc
+             limit 1
+        """
+
+  private def findNextTaskByIdQ(taskId: ObjectId) =
+    s"""
+        select $columns
+        from webknossos.tasks_
+        where _id = '${taskId.id}' and openInstances > 0
+        limit 1
+        """
 
   def assignNext(userId: ObjectId,
                  teamIds: List[ObjectId],
-                 isTeamManagerOrAdmin: Boolean = false): Fox[(Task, ObjectId)] = {
+                 isTeamManagerOrAdmin: Boolean = false): Fox[(ObjectId, ObjectId)] = {
 
     val annotationId = ObjectId.generate
+    val now = new java.sql.Timestamp(System.currentTimeMillis)
```

Comment:
If the related column is of type timestamp, we might directly use `NOW()` or `LOCALTIMESTAMP()` or sth similar, but probably doesn't matter with those queries.